arangod/RestHandler/RestCursorHandler.cpp

```diff
@@ -618,10 +623,24 @@ RestStatus RestCursorHandler::generateCursorResult(rest::ResponseCode code) {
     builder.add(StaticStrings::Error, VPackValue(false));
     builder.add(StaticStrings::Code, VPackValue(static_cast<int>(code)));
     builder.close();
+    _cursor->setLastQueryBatchObject(builder);
 
     _response->setContentType(rest::ContentType::JSON);
+    TRI_IF_FAILURE("MakeConnectionErrorForRetry") { return RestStatus::FAIL; }
+
     generateResult(code, std::move(buffer), std::move(ctx));
   } else {
+    VPackBuilder builder;
```

Comment:
We seem to already have a Builder object on the stack here.
Why do we need yet another one?